blocks/main_loop.py

```diff
@@ -105,6 +106,10 @@ def run(self):
                 pass
         except TrainingFinish:
             self.log.current_row.training_finished = True
+        except Exception as e:
+            logger.error(traceback.format_exc(e))
+            logger.info("Still trying to finish gracefully")
```

Comment:
My idea was that they should have seen the ERROR right above. But okay, I will change that.